test/e2e/framework/device_plugin_utils.go

```diff
@@ -0,0 +1,221 @@
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha"
+	"k8s.io/kubernetes/test/e2e_node/builder"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+const (
+	// dummy device file saved in "/tmp/dummy-device" directory,
+	// e.g. "/tmp/dummy-device/device-1", "/tmp/dummy-device/device-2"
+	DummyDeviceDir string = "dummy-device"
+	DummyDeviceRE  string = "^device-[0-9]*$"
+)
+
+var (
+	// Dummy device resource name to be advertised to Kubelet
+	ResourceName = "resource.name/1"
+)
+
+func NumberOfStubDevices(node *v1.Node) int64 {
+	val, ok := node.Status.Capacity[v1.ResourceName(ResourceName)]
+
+	if !ok {
+		return 0
+	}
+
+	return val.Value()
+}
+
+// StubDevicePlugin returns the stub Device Plugin pod for conformance testing
+func StubDevicePlugin(ns string) *v1.Pod {
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stub-device-plugin-" + string(uuid.NewUUID()),
+			Namespace: ns,
+		},
+
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyAlways,
+			Volumes: []v1.Volume{{
+				Name: "device-plugin-path",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: pluginapi.DevicePluginPath,
+					},
+				},
+			}, {
+				Name: "dummy-device-path",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/tmp/dummy-device",
+					},
+				},
+			}},
+			Containers: []v1.Container{
+				{
+					Name:  "stub-device-plugin",
+					Image: imageutils.GetE2EImage(imageutils.StubDevicePlugin),
+					// Args:  []string{"-resource_name", ResourceName},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "device-plugin-path",
+							MountPath: pluginapi.DevicePluginPath,
+						},
+						{
+							Name:      "dummy-device-path",
+							MountPath: "/tmp/dummy-device",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return p
+}
+
+// DummyDevice represents the dummy devices details with resource name.
+type DummyDevice struct {
+	ResourceName string             `json:"resourceName,omitempty"`
+	Devices      []pluginapi.Device `json:"devices,omitempty"`
+}
+
+// DummyDeviceList is a list of DummyDevice objects.
+type DummyDeviceList struct {
+	Items []DummyDevice `json:"items,omitempty"`
+}
+
+// CreateDummyDeviceFileFromConfig reads config file and parse it to create dummy device files on node
+// dummy device directory looks like this:
+// |-- tmp
+//     |-- dummy-device
+//         |-- resource.name-1  # hacky here, replace "-" with "/" to follow kubernetes resource naming convention
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: unhealthy)
+//         |-- resource.name-2
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: healthy)
+//             |-- device-3 (content: unhealthy)
+//         |-- resource.name-3
+//             |-- device-1 (content: healthy)
+//             |-- device-2 (content: healthy)
+func CreateDummyDeviceFileFromConfig() error {
```

Comment:
Can we call this file from stub device plugin instead of from the test?

Reply:
I think stub device plugin is a `slave/worker`, it will read/watch these `dummy device files` of a reaource.
Test is a `master/manager` which can create/write `dummy device files` with device state in file content from `config file`.
This maybe more flexible to extend the test case.